src/windows/models/properties_model.py

```diff
@@ -416,163 +419,152 @@ def value_updated(self, item, interpolation=-1, value=None, interpolation_detail
+                        point["co"]["X"], interpolation,
+                        str(interpolation_details[2:4]))
+
+            # Delete point (if needed)
+            if point_to_delete:
+                clip_updated = True
+                log.debug("Found point to delete at X=%s" % point_to_delete["co"]["X"])
+                d[property_key]["Points"].remove(point_to_delete)
+
+            # Create new point (if needed)
+            elif not found_point and new_value is not None:
+                clip_updated = True
+                log.debug("Created new point at X=%d", self.frame_number)
+                d[property_key]["Points"].append({
+                    'co': {'X': self.frame_number, 'Y': new_value},
+                    'interpolation': 1})
+
+        if not clip_updated:
+            # If no keyframe was found, set a basic property
+            if property_type == "int":
+                clip_updated = True
+                try:
+                    d[property_key] = int(new_value)
+                except Exception as ex:
+                    log.warn('Invalid Integer value passed to property: %s' % ex)
+
+            elif property_type == "float":
+                clip_updated = True
+                try:
+                    d[property_key] = float(new_value)
+                except Exception as ex:
+                    log.warn('Invalid Float value passed to property: %s' % ex)
 
-                elif property_type in ["font", "caption"]:
-                    clip_updated = True
-                    try:
-                        clip_data[property_key] = str(new_value)
-                    except Exception as ex:
-                        log.warn('Invalid Font/Caption value passed to property: %s' % ex)
+            elif property_type == "bool":
+                clip_updated = True
+                try:
+                    d[property_key] = bool(new_value)
+                except Exception as ex:
+                    log.warn('Invalid Boolean value passed to property: %s' % ex)
 
-                elif property_type == "reader":
-                    # Transition
-                    clip_updated = True
-                    try:
-                        clip_object = openshot.Clip(value)
-                        clip_object.Open()
-                        clip_data[property_key] = json.loads(clip_object.Reader().Json())
-                        clip_object.Close()
-                        clip_object = None
-                    except Exception as ex:
-                        log.warn('Invalid Reader value passed to property: %s (%s)' % (value, ex))
-
-            # Reduce # of clip properties we are saving (performance boost)
-            clip_data = {property_key: clip_data.get(property_key)}
-            if object_id:
-                clip_data = {'objects': {object_id: clip_data}}
-
-            # Save changes
-            if clip_updated:
-                # Save
-                c.save()
```

Comment:
_Fortunately_ (in a sense), the reassignment of `clip_data` here broke that reference, and created a new temporary object instead of modifying `c.data` to only contain the changed property. The `c` object therefore still contained the entire property set, without optimization. Then, `c.save()` sent that full, modified property list to the update manager.